### ISSUE TITLE
RPC failing to write in Memory extended games

### ISSUE THREAD

**Issue by NimbusFox** [citra-emu/citra#6043]
Commit hash: master-5a7d80172 (This has occurred in previous versions too since 1604)

When accessing FCRAM+0 (Documented here: https://www.3dbrew.org/wiki/Memory_layout#IO_Process_virtual_addressing_equivalence) in newer 3DS titles like Pokemon Gen 7 (Sun/Moon/Ultra Sun/Ultra Moon) the RPC is able to read from this section of memory but fails to write.

I have tried with Gen 6 Pokemon Games (X/Y/Omega Ruby/Alpha Sapphire) and I can write to memory just fine modifying the Pokemon. But with Gen 7 it does not write to memory. Gen 7 party data is stored in 0x34195E10 (Sun/Moon) and 0x33F7FA44 (Ultra Sun/Ultra Moon) where as gen 6 stores them in 0x08CE1CE8 (X/Y 1.0) and 0x08CF727C (Omega Ruby/Alpha Sapphire 1.0)

I hope we can fix this in the future as a tool I am working on unfortunately would have to miss out on planned features from working with Gen 7 Pokemon games

# TL;DR
RPC can read in memory sections 0x30000000 - 0x40000000 but can't write to them (Newer 3DS games used this space for memory)

but can read and write in memory sections 0x08000000 - 0x10000000 (Older 3DS games used this space for memory)

**Comment by SachinVin** [comment]
Seems like writes are restricted to only [these](https://github.com/citra-emu/citra/blob/master/src/core/rpc/rpc_server.cpp#L41-L44) regions, the original PR doesn't have any documentation as to why the linear heap regions are ignored. (possibly due to being shared by the GPU?)

**Comment by SachinVin** [comment]
Reads are unrestricted only the writes are limited to certain areas

**Comment by NimbusFox** [comment]
Would there be any chance that restriction could be lifted for writing in that section?
